echo-pipelinetriggers/src/main/java/com/netflix/spinnaker/echo/pipelinetriggers/monitor/ManualEventMonitor.java

```diff
@@ -0,0 +1,123 @@
+import org.springframework.stereotype.Component;
+import rx.functions.Action1;
+import rx.functions.Func1;
+
+/**
+ * Triggers pipelines in _Orca_ when a user manually starts a pipeline.
+ */
+@Slf4j
+@Component
+public class ManualEventMonitor extends TriggerMonitor {
+
+  public static final String MANUAL_TRIGGER_TYPE = "manual";
+
+  @Override
+  protected boolean handleEventType(String eventType) {
+    return eventType.equalsIgnoreCase(ManualEvent.TYPE);
+  }
+
+  @Override
+  protected ManualEvent convertEvent(Event event) {
+    return objectMapper.convertValue(event, ManualEvent.class);
+  }
+
+  public ManualEventMonitor(@NonNull PipelineCache pipelineCache,
+    @NonNull Action1<Pipeline> subscriber,
+    @NonNull Registry registry) {
+    super(pipelineCache, subscriber, registry);
+  }
+
+  protected Func1<Pipeline, Optional<Pipeline>> withMatchingTrigger(final TriggerEvent event) {
+    return pipeline -> {
+      if (pipeline.isDisabled()) {
+        return Optional.empty();
+      } else {
+        ManualEvent manualEvent = (ManualEvent) event;
+        Trigger trigger = manualEvent.getContent().getTrigger();
+        return Stream.of(trigger)
+          .filter(matchTriggerFor(event, pipeline))
+          .findFirst()
+          .map(buildTrigger(pipeline, event));
+      }
+    };
+  }
+
+  @Override
+  protected boolean isSuccessfulTriggerEvent(final TriggerEvent event) {
+    return true;
+  }
+
+  @Override
+  protected Function<Trigger, Pipeline> buildTrigger(Pipeline pipeline, TriggerEvent event) {
+    ManualEvent manualEvent = (ManualEvent) event;
+    return trigger -> {
+      String runAsUser = ((ManualEvent) event).getContent().getRunAsUser();
+      List<Map<String, Object>> notifications = buildNotifications(pipeline.getNotifications(),
+        manualEvent.getContent().getTrigger().getNotifications());
+      return pipeline
+        .withTrigger(manualEvent.getContent().getTrigger().atRunAsUser(runAsUser))
+        .withNotifications(notifications);
+    };
+  }
+
+  private List<Map<String, Object>> buildNotifications(List<Map<String, Object>> pipelineNotifications, List<Map<String, Object>> triggerNotifications) {
+    List<Map<String, Object>> notifications = new ArrayList<>();
+    if (pipelineNotifications != null) {
+      notifications.addAll(pipelineNotifications);
+    }
+    if (triggerNotifications != null) {
+      notifications.addAll(triggerNotifications);
+    }
+    return notifications;
+  }
+
+  @Override
+  protected boolean isValidTrigger(final Trigger trigger) {
+    return true;
+  }
+
+  @Override
+  protected Predicate<Trigger> matchTriggerFor(final TriggerEvent event, final Pipeline pipeline) {
+    ManualEvent manualEvent = (ManualEvent) event;
+    String application = manualEvent.getContent().getApplication();
+    String nameOrId = manualEvent.getContent().getPipelineNameOrId();
+
+    return trigger -> pipeline.getApplication().equals(application) && (
```

Comment:
I'd left it as is to match the current behavior, which is case-sensitive.  (Currently when Gate looks up the pipeline you specified from front50, it will 404 unless there's a case-sensitive match.)